CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -380,7 +380,6 @@ public static function checkFinancialTypeHasDeferred($params, $contributionID =
       return FALSE;
```

Comment:
stylistically, I'd prefer to see a test around $params['prevContribution']->revenue_recognition_date), eg !CRM_Utils_System::isNull($params['prevContribution']->revenue_recognition_date)) 
Also, same for around $contributionID